src/other/contango/ContangoWand.sol

```diff
@@ -0,0 +1,317 @@
+        if (series_.fyToken != IFYToken(address(0))) {
+            (IPool pool_,) = yieldSpaceOracle.sources(baseId, ilkId);
+            require(address(pool_) != address(0), "YieldSpace oracle not set");
+            compositeOracle.setSource(baseId, ilkId, yieldSpaceOracle);
+        } else if (yieldCauldron.assets(ilkId) != address(0)) {
+            DataTypes.SpotOracle memory spotOracle_ = yieldCauldron.lookupSpotOracle(baseId, ilkId);
+            compositeOracle.setSource(baseId, ilkId, spotOracle_.oracle);
+        }
+    }
+
+    /// @notice Set a path in the Composite oracle, as long as the path is not overwriting anything
+    function setCompositeOraclePath(bytes6 baseId, bytes6 quoteId, bytes6[] calldata path) external auth {
+        // This is hideous, but's the only way to check if a path is already set
+        try compositeOracle.paths(baseId, quoteId, 0) {
+            revert("Path already set");
+        } catch {
+            compositeOracle.setPath(baseId, quoteId, path);
+        }
+    }
+
+    /// ----------------- Ladle Governance -----------------
+
+    /// @notice Propagate a pool to the Ladle from the Yield Ladle
+    function addPool(bytes6 seriesId) external auth returns (IPool pool_) {
+        pool_ = IPool(yieldLadle.pools(seriesId));
+        require(address(pool_) != address(0), "Pool not known to the Yield Ladle");
+        contangoLadle.addPool(seriesId, pool_);
+    }
+
+    /// @notice Propagate an integration to the Ladle from the Yield Ladle
+    function addIntegration(address integration) external auth {
+        contangoLadle.addIntegration(integration, yieldLadle.integrations(integration));
+    }
+
+    /// @notice Propagate a token to the Ladle from the Yield Ladle
+    function addToken(address token) external auth {
+        contangoLadle.addToken(token, yieldLadle.tokens(token));
+    }
+
+    function copyJoin(bytes6 assetId) external auth {
+        _copyJoin(assetId);
+    }
+
+    function _copyJoin(bytes6 assetId) internal {
+        IJoin join = yieldLadle.joins(assetId);
+        require(address(join) != address(0), "Join not known to the Yield Ladle");
+        _addJoin(assetId, yieldLadle.joins(assetId));
+    }
+
+    function _addJoin(bytes6 assetId, IJoin join) internal {
+        contangoLadle.addJoin(assetId, join);
+
+        AccessControl(address(join)).grantRole(IJoin.join.selector, address(contangoLadle));
+        AccessControl(address(join)).grantRole(IJoin.exit.selector, address(contangoLadle));
+
+        AccessControl(address(join)).grantRole(IJoin.join.selector, address(contangoWitch));
+        AccessControl(address(join)).grantRole(IJoin.exit.selector, address(contangoWitch));
+    }
+
+    // TODO Maybe check if the join exists before deploying?
+
+    function deployJoin(bytes6 assetId) external auth returns (IJoin join) {
+        address asset = contangoCauldron.assets(assetId);
+        require(asset != address(0), "Asset not known to the Contango Cauldron");
```

Comment:
`require(yieldLadle.joins(assetId) == IJoin(address(0)), "Join already exists")`.

Reply:
good point

Reply:
btw, here I'm trying to avoid overriding an existing join, but I agree we should also check there isn't one that we could copy from the yieldLadle